src/main/java/org/tugraz/sysds/runtime/instructions/cp/SpoofCPInstruction.java

```diff
@@ -96,8 +97,17 @@ else if (output.getDataType() == DataType.SCALAR) {
 	}
 	
 	@Override
-	public LineageItem[] getLineageItems(ExecutionContext ec) {
-		return new LineageItem[]{new LineageItem(output.getName(),
-			getOpcode(), LineageItemUtils.getLineage(ec, _in))};
+	public LineageItem[] getLineageItems(ExecutionContext ec) 
+	{
+		//read and deepcopy the corresponding lineage DAG (pre-codegen)
+		LineageItem LIroot = LineageCodegenItem.getCodegenLTrace(this.getOperatorClass().getName()).DeepCopy();
+
+		//replace the placeholders with original instruction inputs. 
+		LineageItemUtils.replaceDagLeaves(ec, LIroot, _in);
+
+		//replace the placeholder name with original output's name
+		LIroot.setName(output.getName());
```

Comment:
As we discussed offline, we should really get rid of the lineage item names as they are irrelevant. I guess they are only used temporarily when creating new lineage items (to have a simple API)

Reply:
Yes, sure, will do that in a future PR.